pyomo/core/base/indexed_component.py

```diff
@@ -360,15 +360,17 @@ def iteritems(self):
 
     def keys(self):
         """Return an iterator of the keys in the dictionary"""
-        return [ x for x in self ]
+        return iter(self)
 
     def values(self):
         """Return an iterator of the component data objects in the dictionary"""
-        return [ self[x] for x in self ]
+        for s in self:
+            yield self[s]
```

Comment:
You don't need to change this, but why `s` and not `x`?

Reply:
Because it's the first letter of `self` - no other reason.